Rajitha/LoopsTest.js

```diff
@@ -0,0 +1,203 @@
+//write a function that alternates items in each array - [1,6,2,7,3,8,4,9,5,10]
+******************************************************************************/
+const arr1 = [6,7,8,9,10];
+
+const combineArray = (arr,arr1) => (arr.concat(arr1))
+    //return newArray = arr.concat(arr1); //Is it correct???
+//};
+console.log(combineArray(arr,arr1));
+
+const alternateArray = (arr,arr1) => {
+    let altArray = [];
+    for (let i=0; i<arr.length; i++) {
+        altArray = altArray.concat(arr[i], arr1[i]);
+    }
+    return altArray;
+}
+console.log(alternateArray(arr,arr1));
+
+/**************************************************************
+ * write a function that returns all even number - [2,8,10,6]
+ * write a function that returns all odd numbers - [1,5]
+ * **************************************************************/
+const arr2 = [2,8,1,10,5,6];
+
+const evenOddArrays = arr => {
+    let evenArray = [];
+    let oddArray = [];
+    for (let i=0; i<arr.length; i++) {
+        if (arr[i]%2 == 0) {
+            evenArray.push(arr[i]);
+        } else {
+            oddArray.push(arr[i]);
+        }
+    }
+    console.log(evenArray);
+    console.log(oddArray);
+    return [evenArray, oddArray];
+};
+
+console.log(evenOddArrays(arr2)); //is there a way to return two arrays in a single function? multi-dimensinal array?
+
+/**************************************************************
+ * Using DO loop
+ * write a function that returns all even number - [2,8,10,6]
+ * write a function that returns all odd numbers - [1,5]
+ * **************************************************************/
+const evenArrayDoLoop = arr => {
+    let evenArray = [];
+    let i = 0;
+    do {
+        if (arr[i]%2 == 0) {
+            evenArray.push(arr[i]);
+        }
+        i++;
+    }
+    while (i<=arr.length);
+    return evenArray;
+}
+console.log(evenArrayDoLoop(arr2));
+
+/******* Do Loop - Odd Array ********/
+ const oddArrayDoLoop = arr => {
+    let oddArray = [];
+    let i = 0;
+    do {
+        if (arr[i]%2 != 0) {
+            oddArray.push(arr[i]);
+        }
+        i++;
+    }
+    while (i<=arr.length); //Why does it return undefined here but not in evenDoLoop????
+    return oddArray;
+}
+ console.log(oddArrayDoLoop(arr2));
+
+var arr3 = [
+    {type: "socks", quantity: 1},
+    {type: "shoes", quantity: 3},
+    {type: "pants", quantity: 1},
+    {type: "socks", quantity: 5},
+    {type: "shoes", quantity: 1},
+    {type: "socks", quantity: 10}
+];
+
+/******* Write a function that returns the total quantity of the socks ****/
+const totalSocks = arr => {
+    let socks = 0;
+    for (let i=0; i<arr.length; i++){
+        socks += arr[i].quantity;
+    }
+    return socks;
+}
+console.log("Total socks " +totalSocks(arr3));
+
+/***** Using forEach *****/
+let totSocks = 0;
+
+arr3.forEach((addSocks) => {
+    totSocks += addSocks.quantity;
+    console.log(totSocks);
+}); //Can we return the final count of socks???
```

Comment:
You will need to add a return statement to return your final result. Best way for this is to probably wrap your above work within a function and call the function.

Reply:
/***** Using forEach *****/
let totSocks = 0;
function getSocksCount(addSocks){
    arr3.forEach((addSocks) => {
        if (addSocks.type === 'socks') {
            totSocks += addSocks.quantity;
        }
    });
    return totSocks;
};
console.log("Total no. of socks using forEach function " +getSocksCount(arr3));